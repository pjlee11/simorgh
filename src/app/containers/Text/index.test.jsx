import React from 'react';
import {
  shouldShallowMatchSnapshot,
  isNull,
} from '../../helpers/tests/testHelpers';
import TextContainer from './index';

describe('TextContainer', () => {
  describe('with no data', () => {
    isNull('should return null', <TextContainer />);
  });

  describe('with data', () => {
    const fragmentBlock = (text, attributes = []) => ({
      type: 'fragment',
      model: {
        text,
        attributes,
      },
    });

    const paragraphBlock = blocks => ({
      type: 'paragraph',
      model: {
        blocks,
      },
    });

    const data = {
      blocks: [
        paragraphBlock([fragmentBlock('This is a 1st paragraph block.')]),
        paragraphBlock([fragmentBlock('This is a 2nd paragraph block.')]),
        paragraphBlock([fragmentBlock('This is a 3rd paragraph block.')]),
        paragraphBlock([fragmentBlock('This is a 4th paragraph block..')]),
        paragraphBlock([fragmentBlock('This is a 5th paragraph block.')]),
      ],
    };

    shouldShallowMatchSnapshot(
      'should render correctly',
      <TextContainer {...data} />,
    );

    describe('with a passed previous block type', () => {
      shouldShallowMatchSnapshot(
        'should render correctly',
        <TextContainer {...data} typeOfPreviousBlock="text" />,
      );
    });
  });
});
