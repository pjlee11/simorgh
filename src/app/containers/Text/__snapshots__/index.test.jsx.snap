// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextContainer with data should render correctly 1`] = `
<React.Fragment>
  <Blocks
    blocks={
      Array [
        Object {
          "model": Object {
            "blocks": Array [
              Object {
                "model": Object {
                  "attributes": Array [],
                  "text": "This is a 1st paragraph block.",
                },
                "type": "fragment",
              },
            ],
          },
          "type": "paragraph",
        },
        Object {
          "model": Object {
            "blocks": Array [
              Object {
                "model": Object {
                  "attributes": Array [],
                  "text": "This is a 2nd paragraph block.",
                },
                "type": "fragment",
              },
            ],
          },
          "type": "paragraph",
        },
        Object {
          "model": Object {
            "blocks": Array [
              Object {
                "model": Object {
                  "attributes": Array [],
                  "text": "This is a 3rd paragraph block.",
                },
                "type": "fragment",
              },
            ],
          },
          "type": "paragraph",
        },
        Object {
          "model": Object {
            "blocks": Array [
              Object {
                "model": Object {
                  "attributes": Array [],
                  "text": "This is a 4th paragraph block..",
                },
                "type": "fragment",
              },
            ],
          },
          "type": "paragraph",
        },
        Object {
          "model": Object {
            "blocks": Array [
              Object {
                "model": Object {
                  "attributes": Array [],
                  "text": "This is a 5th paragraph block.",
                },
                "type": "fragment",
              },
            ],
          },
          "type": "paragraph",
        },
      ]
    }
    componentsToRender={
      Object {
        "paragraph": [Function],
      }
    }
  />
</React.Fragment>
`;

exports[`TextContainer with data with a passed previous block type should render correctly 1`] = `
<React.Fragment>
  <hr />
  <Blocks
    blocks={
      Array [
        Object {
          "model": Object {
            "blocks": Array [
              Object {
                "model": Object {
                  "attributes": Array [],
                  "text": "This is a 1st paragraph block.",
                },
                "type": "fragment",
              },
            ],
          },
          "type": "paragraph",
        },
        Object {
          "model": Object {
            "blocks": Array [
              Object {
                "model": Object {
                  "attributes": Array [],
                  "text": "This is a 2nd paragraph block.",
                },
                "type": "fragment",
              },
            ],
          },
          "type": "paragraph",
        },
        Object {
          "model": Object {
            "blocks": Array [
              Object {
                "model": Object {
                  "attributes": Array [],
                  "text": "This is a 3rd paragraph block.",
                },
                "type": "fragment",
              },
            ],
          },
          "type": "paragraph",
        },
        Object {
          "model": Object {
            "blocks": Array [
              Object {
                "model": Object {
                  "attributes": Array [],
                  "text": "This is a 4th paragraph block..",
                },
                "type": "fragment",
              },
            ],
          },
          "type": "paragraph",
        },
        Object {
          "model": Object {
            "blocks": Array [
              Object {
                "model": Object {
                  "attributes": Array [],
                  "text": "This is a 5th paragraph block.",
                },
                "type": "fragment",
              },
            ],
          },
          "type": "paragraph",
        },
      ]
    }
    componentsToRender={
      Object {
        "paragraph": [Function],
      }
    }
  />
</React.Fragment>
`;
